scripts/check_formatting

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+
+# Runs clang-format and rustfmt on the entire codebase to check if it's 
+# well formatted. Exit codes:
+#  - 1 there are files to be formatted.
+#  - 0 everything looks fine.
+
+# We ask clang-format to generate an XML output with the deisred formattings.
+# We check if any "replacement" tag exists in the output.
+CLANG_FORMAT_EXIT_CODE=0
+find oak examples -type f -name '*.h' -o -name '*.cc' -o -name '*.proto' \
+  | xargs -I{} clang-format -style=file -output-replacements-xml {} \
```

Comment:
Alternatively you could use `find -exec` instead of `xargs`, but up to you really.

Reply:
For some reason I've always preferred xargs to -exec with find.

Reply:
SG! It seems at least the "simple" version of `xargs` could be used though? i.e. `xargs clang-format -style=file -output-replacements-xml` (no explicit `-I{}`).